Request: Add NSTextField class

Please add support for NSTextField. Thank you.

You are probably talking about using objc.NewClass(). I did make this example program as a start but for some reason it keeps panicking. Do you see anything wrong with it?

```
package main

import "github.com/progrium/macdriver/cocoa"
import "github.com/progrium/macdriver/objc"
import "fmt"

func main() {
	app := cocoa.NSApp()
	cls := objc.NewClass("NSTextField", "NSControl")
	cls.AddMethod("labelWithString:", func(message string) {
		fmt.Println("Message:", message)
	})
	objc.RegisterClass(cls)
	app.Run()
}
```

Here is the panic:
panic: unable to AllocateClassPair

I traced the error to objc/class.go to this code:

```
func NewClass(classname string, superclass string) Class {
	superClass := GetClass(superclass)

	ptr := allocateClassPair(unsafe.Pointer(superClass.Pointer()), classname)
	if ptr == nil {
		panic("unable to AllocateClassPair")
	}

	// Register the dealloc method to be able to properly remove the classInfo
	// reference to our internal pointer.
	classAddMethod(ptr, "dealloc", encVoid+encId+encSelector)

	lazilyRegisterClassInMap(classname)
	return object{ptr: uintptr(ptr)}
}
```

It looks like allocateClassPair() is the problem. For some reason it doesn't like NSTextField.

--- a/api/appkit/nstextfield.objc.json
+++ b/api/appkit/nstextfield.objc.json
@@ -1,0 +1,450 @@
+{
+  "Class": {
+    "Name": "NSTextField",
+    "Description": "Text the user can select or edit to send an action message to a target when the user presses the Return key.",
+    "Declaration": "@interface NSTextField : NSControl",
+    "Platforms": [
+      "macOS 10.0+"
+    ],
+    "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield?language=objc",
+    "InstanceMethods": [
+      {
+        "Name": "selectText:",
+        "Description": "Ends editing in the text field and, if it’s selectable, selects the entire text content.",
+        "Declaration": "- (void)selectText:(id)sender;",
+        "Return": {
+          "Name": "void"
+        },
+        "Args": [
+          {
+            "Name": "sender",
+            "Type": {
+              "Name": "id"
+            }
+          }
+        ],
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/1399430-selecttext?language=objc"
+      },
+      {
+        "Name": "setTitleWithMnemonic:",
+        "Description": "Sets the text field’s string value using the embedded character as the keyboard mnemonic.",
+        "Declaration": "- (void)setTitleWithMnemonic:(NSString *)stringWithAmpersand;",
+        "Return": {
+          "Name": "void"
+        },
+        "Args": [
+          {
+            "Name": "stringWithAmpersand",
+            "Type": {
+              "Name": "NSString",
+              "IsPtr": true
+            }
+          }
+        ],
+        "Deprecated": true,
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/1399411-settitlewithmnemonic?language=objc"
+      },
+      {
+        "Name": "textShouldBeginEditing:",
+        "Description": "Requests permission to begin editing a text object.",
+        "Declaration": "- (BOOL)textShouldBeginEditing:(NSText *)textObject;",
+        "Return": {
+          "Name": "BOOL"
+        },
+        "Args": [
+          {
+            "Name": "textObject",
+            "Type": {
+              "Name": "NSText",
+              "IsPtr": true
+            }
+          }
+        ],
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/1399399-textshouldbeginediting?language=objc"
+      },
+      {
+        "Name": "textDidBeginEditing:",
+        "Description": "Posts a notification to the default notification center that the text is about to go into edit mode.",
+        "Declaration": "- (void)textDidBeginEditing:(NSNotification *)notification;",
+        "Return": {
+          "Name": "void"
+        },
+        "Args": [
+          {
+            "Name": "notification",
+            "Type": {
+              "Name": "NSNotification",
+              "IsPtr": true
+            }
+          }
+        ],
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/1399426-textdidbeginediting?language=objc"
+      },
+      {
+        "Name": "textDidChange:",
+        "Description": "Posts a notification when the text changes, and forwards the message to the text field’s cell if it responds.",
+        "Declaration": "- (void)textDidChange:(NSNotification *)notification;",
+        "Return": {
+          "Name": "void"
+        },
+        "Args": [
+          {
+            "Name": "notification",
+            "Type": {
+              "Name": "NSNotification",
+              "IsPtr": true
+            }
+          }
+        ],
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/1399397-textdidchange?language=objc"
+      },
+      {
+        "Name": "textShouldEndEditing:",
+        "Description": "Performs validation on the text field’s new value.",
+        "Declaration": "- (BOOL)textShouldEndEditing:(NSText *)textObject;",
+        "Return": {
+          "Name": "BOOL"
+        },
+        "Args": [
+          {
+            "Name": "textObject",
+            "Type": {
+              "Name": "NSText",
+              "IsPtr": true
+            }
+          }
+        ],
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/1399434-textshouldendediting?language=objc"
+      },
+      {
+        "Name": "textDidEndEditing:",
+        "Description": "Posts a notification when the text is no longer in edit mode.",
+        "Declaration": "- (void)textDidEndEditing:(NSNotification *)notification;",
+        "Return": {
+          "Name": "void"
+        },
+        "Args": [
+          {
+            "Name": "notification",
+            "Type": {
+              "Name": "NSNotification",
+              "IsPtr": true
+            }
+          }
+        ],
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/1399420-textdidendediting?language=objc"
+      }
+    ],
+    "InstanceProperties": [
+      {
+        "Name": "selectable",
+        "Description": "A Boolean value that determines whether the user can select the content of the text field.",
+        "Declaration": "@property(getter=isSelectable) BOOL selectable;",
+        "Type": {
+          "Name": "BOOL"
+        },
+        "Attrs": {
+          "getter": "isSelectable"
+        },
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/1399422-selectable?language=objc"
+      },
+      {
+        "Name": "editable",
+        "Description": "A Boolean value that controls whether the user can edit the value in the text field.",
+        "Declaration": "@property(getter=isEditable) BOOL editable;",
+        "Type": {
+          "Name": "BOOL"
+        },
+        "Attrs": {
+          "getter": "isEditable"
+        },
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/1399407-editable?language=objc"
+      },
+      {
+        "Name": "allowsEditingTextAttributes",
+        "Description": "A Boolean value that controls whether the user can change font attributes of the text field’s string.",
+        "Declaration": "@property BOOL allowsEditingTextAttributes;",
+        "Type": {
+          "Name": "BOOL"
+        },
+        "Attrs": {},
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/1399401-allowseditingtextattributes?language=objc"
+      },
+      {
+        "Name": "importsGraphics",
+        "Description": "A Boolean value that controls whether the user can drag image files into the text field.",
+        "Declaration": "@property BOOL importsGraphics;",
+        "Type": {
+          "Name": "BOOL"
+        },
+        "Attrs": {},
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/1399428-importsgraphics?language=objc"
+      },
+      {
+        "Name": "placeholderString",
+        "Description": "The string the text field displays when empty to help the user understand the text field’s purpose.",
+        "Declaration": "@property(copy) NSString *placeholderString;",
+        "Type": {
+          "Name": "NSString",
+          "IsPtr": true
+        },
+        "Attrs": {
+          "copy": true
+        },
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/1399391-placeholderstring?language=objc"
+      },
+      {
+        "Name": "placeholderAttributedString",
+        "Description": "The attributed string the text field displays when empty to help the user understand the text field’s purpose.",
+        "Declaration": "@property(copy) NSAttributedString *placeholderAttributedString;",
+        "Type": {
+          "Name": "NSAttributedString",
+          "IsPtr": true
+        },
+        "Attrs": {
+          "copy": true
+        },
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/1399387-placeholderattributedstring?language=objc"
+      },
+      {
+        "Name": "lineBreakStrategy",
+        "Description": "The strategy that the system uses to break lines when laying out multiple lines of text.",
+        "Declaration": "@property NSLineBreakStrategy lineBreakStrategy;",
+        "Type": {
+          "Name": "NSLineBreakStrategy"
+        },
+        "Attrs": {},
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/3667464-linebreakstrategy?language=objc"
+      },
+      {
+        "Name": "allowsDefaultTighteningForTruncation",
+        "Description": "A Boolean value that controls whether single-line text fields tighten intercharacter spacing before truncating the text.",
+        "Declaration": "@property BOOL allowsDefaultTighteningForTruncation;",
+        "Type": {
+          "Name": "BOOL"
+        },
+        "Attrs": {},
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/1399405-allowsdefaulttighteningfortrunca?language=objc"
+      },
+      {
+        "Name": "maximumNumberOfLines",
+        "Description": "The maximum number of lines a wrapping text field displays before clipping or truncating the text.",
+        "Declaration": "@property NSInteger maximumNumberOfLines;",
+        "Type": {
+          "Name": "NSInteger"
+        },
+        "Attrs": {},
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/1399424-maximumnumberoflines?language=objc"
+      },
+      {
+        "Name": "preferredMaxLayoutWidth",
+        "Description": "The maximum width of the text field’s intrinsic content size.",
+        "Declaration": "@property CGFloat preferredMaxLayoutWidth;",
+        "Type": {
+          "Name": "CGFloat"
+        },
+        "Attrs": {},
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/1399395-preferredmaxlayoutwidth?language=objc"
+      },
+      {
+        "Name": "textColor",
+        "Description": "The color of the text field’s content.",
+        "Declaration": "@property(copy) NSColor *textColor;",
+        "Type": {
+          "Name": "NSColor",
+          "IsPtr": true
+        },
+        "Attrs": {
+          "copy": true
+        },
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/1399409-textcolor?language=objc"
+      },
+      {
+        "Name": "backgroundColor",
+        "Description": "The color of the background the text field’s cell draws behind the text.",
+        "Declaration": "@property(copy) NSColor *backgroundColor;",
+        "Type": {
+          "Name": "NSColor",
+          "IsPtr": true
+        },
+        "Attrs": {
+          "copy": true
+        },
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/1399389-backgroundcolor?language=objc"
+      },
+      {
+        "Name": "drawsBackground",
+        "Description": "A Boolean value that controls whether the text field’s cell draws a background color behind the text.",
+        "Declaration": "@property BOOL drawsBackground;",
+        "Type": {
+          "Name": "BOOL"
+        },
+        "Attrs": {},
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/1399416-drawsbackground?language=objc"
+      },
+      {
+        "Name": "bezeled",
+        "Description": "A Boolean value that controls whether the text field draws a bezeled background around its contents.",
+        "Declaration": "@property(getter=isBezeled) BOOL bezeled;",
+        "Type": {
+          "Name": "BOOL"
+        },
+        "Attrs": {
+          "getter": "isBezeled"
+        },
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/1399435-bezeled?language=objc"
+      },
+      {
+        "Name": "bezelStyle",
+        "Description": "The text field’s bezel style, square or rounded.",
+        "Declaration": "@property NSTextFieldBezelStyle bezelStyle;",
+        "Type": {
+          "Name": "NSTextFieldBezelStyle"
+        },
+        "Attrs": {},
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/1399418-bezelstyle?language=objc"
+      },
+      {
+        "Name": "bordered",
+        "Description": "A Boolean value that controls whether the text field draws a solid black border around its contents.",
+        "Declaration": "@property(getter=isBordered) BOOL bordered;",
+        "Type": {
+          "Name": "BOOL"
+        },
+        "Attrs": {
+          "getter": "isBordered"
+        },
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/1399403-bordered?language=objc"
+      },
+      {
+        "Name": "acceptsFirstResponder",
+        "Description": "A Boolean value that indicates whether the text field is editable and accepts first responder status.",
+        "Declaration": "@property(readonly) BOOL acceptsFirstResponder;",
+        "Type": {
+          "Name": "BOOL"
+        },
+        "Attrs": {
+          "readonly": true
+        },
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/1399393-acceptsfirstresponder?language=objc"
+      },
+      {
+        "Name": "allowsCharacterPickerTouchBarItem",
+        "Description": "A Boolean value that controls whether the Touch Bar displays the character picker item for rich text fields.",
+        "Declaration": "@property BOOL allowsCharacterPickerTouchBarItem;",
+        "Type": {
+          "Name": "BOOL"
+        },
+        "Attrs": {},
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/2539553-allowscharacterpickertouchbarite?language=objc"
+      },
+      {
+        "Name": "automaticTextCompletionEnabled",
+        "Description": "A Boolean value that indicates whether the text field automatically completes text as the user types.",
+        "Declaration": "@property(getter=isAutomaticTextCompletionEnabled) BOOL automaticTextCompletionEnabled;",
+        "Type": {
+          "Name": "BOOL"
+        },
+        "Attrs": {
+          "getter": "isAutomaticTextCompletionEnabled"
+        },
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/2539554-automatictextcompletionenabled?language=objc"
+      },
+      {
+        "Name": "delegate",
+        "Description": "The text field’s delegate.",
+        "Declaration": "@property(weak) id\u003cNSTextFieldDelegate\u003e delegate;",
+        "Type": {
+          "Name": "id",
+          "Params": [
+            {
+              "Name": "NSTextFieldDelegate"
+            }
+          ]
+        },
+        "Attrs": {
+          "weak": true
+        },
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/1399437-delegate?language=objc"
+      }
+    ],
+    "TypeMethods": [
+      {
+        "Name": "labelWithAttributedString:",
+        "Description": "Creates a text field for use as a static label that displays styled text, doesn’t wrap, and doesn’t have selectable text.",
+        "Declaration": "+ (instancetype)labelWithAttributedString:(NSAttributedString *)attributedStringValue;",
+        "Return": {
+          "Name": "instancetype"
+        },
+        "Args": [
+          {
+            "Name": "attributedStringValue",
+            "Type": {
+              "Name": "NSAttributedString",
+              "IsPtr": true
+            }
+          }
+        ],
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/1644658-labelwithattributedstring?language=objc"
+      },
+      {
+        "Name": "labelWithString:",
+        "Description": "Initializes a text field for use as a static label that uses the system default font, doesn’t wrap, and doesn’t have selectable text.",
+        "Declaration": "+ (instancetype)labelWithString:(NSString *)stringValue;",
+        "Return": {
+          "Name": "instancetype"
+        },
+        "Args": [
+          {
+            "Name": "stringValue",
+            "Type": {
+              "Name": "NSString",
+              "IsPtr": true
+            }
+          }
+        ],
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/1644377-labelwithstring?language=objc"
+      },
+      {
+        "Name": "textFieldWithString:",
+        "Description": "Initializes a single-line editable text field for user input using the system default font and standard visual appearance.",
+        "Declaration": "+ (instancetype)textFieldWithString:(NSString *)stringValue;",
+        "Return": {
+          "Name": "instancetype"
+        },
+        "Args": [
+          {
+            "Name": "stringValue",
+            "Type": {
+              "Name": "NSString",
+              "IsPtr": true
+            }
+          }
+        ],
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/1644706-textfieldwithstring?language=objc"
+      },
+      {
+        "Name": "wrappingLabelWithString:",
+        "Description": "Initializes a text field for use as a multiline static label with selectable text that uses the system default font.",
+        "Declaration": "+ (instancetype)wrappingLabelWithString:(NSString *)stringValue;",
+        "Return": {
+          "Name": "instancetype"
+        },
+        "Args": [
+          {
+            "Name": "stringValue",
+            "Type": {
+              "Name": "NSString",
+              "IsPtr": true
+            }
+          }
+        ],
+        "TopicURL": "https://developer.apple.com/documentation/appkit/nstextfield/1644543-wrappinglabelwithstring?language=objc"
+      }
+    ]
+  },
+  "APICollection": null,
+  "Kind": "class",
+  "PullDate": "2023-07-05T10:42:39.176034-07:00",
+  "Version": 2
+}

--- a/cocoa/cocoa_objc.gen.go
+++ b/cocoa/cocoa_objc.gen.go
@@ -419,6 +419,26 @@ void* NSText_type_alloc() {
 	return [NSText
 		alloc];
 }
+void* NSTextField_type_alloc() {
+	return [NSTextField
+		alloc];
+}
+void* NSTextField_type_labelWithAttributedString_(void* attributedStringValue) {
+	return [NSTextField
+		labelWithAttributedString: attributedStringValue];
+}
+void* NSTextField_type_labelWithString_(void* stringValue) {
+	return [NSTextField
+		labelWithString: stringValue];
+}
+void* NSTextField_type_textFieldWithString_(void* stringValue) {
+	return [NSTextField
+		textFieldWithString: stringValue];
+}
+void* NSTextField_type_wrappingLabelWithString_(void* stringValue) {
+	return [NSTextField
+		wrappingLabelWithString: stringValue];
+}
 void* NSTextContainer_type_alloc() {
 	return [NSTextContainer
 		alloc];
@@ -3546,6 +3566,201 @@ void* NSText_inst_delegate(void *id) {
 
 void NSText_inst_setDelegate_(void *id, void* value) {
 	[(NSText*)id
+		setDelegate: value];
+}
+
+void NSTextField_inst_selectText_(void *id, void* sender) {
+	[(NSTextField*)id
+		selectText: sender];
+}
+
+BOOL NSTextField_inst_textShouldBeginEditing_(void *id, void* textObject) {
+	return [(NSTextField*)id
+		textShouldBeginEditing: textObject];
+}
+
+BOOL NSTextField_inst_textShouldEndEditing_(void *id, void* textObject) {
+	return [(NSTextField*)id
+		textShouldEndEditing: textObject];
+}
+
+void* NSTextField_inst_init(void *id) {
+	return [(NSTextField*)id
+		init];
+}
+
+BOOL NSTextField_inst_isSelectable(void *id) {
+	return [(NSTextField*)id
+		isSelectable];
+}
+
+void NSTextField_inst_setSelectable_(void *id, BOOL value) {
+	[(NSTextField*)id
+		setSelectable: value];
+}
+
+BOOL NSTextField_inst_isEditable(void *id) {
+	return [(NSTextField*)id
+		isEditable];
+}
+
+void NSTextField_inst_setEditable_(void *id, BOOL value) {
+	[(NSTextField*)id
+		setEditable: value];
+}
+
+BOOL NSTextField_inst_allowsEditingTextAttributes(void *id) {
+	return [(NSTextField*)id
+		allowsEditingTextAttributes];
+}
+
+void NSTextField_inst_setAllowsEditingTextAttributes_(void *id, BOOL value) {
+	[(NSTextField*)id
+		setAllowsEditingTextAttributes: value];
+}
+
+BOOL NSTextField_inst_importsGraphics(void *id) {
+	return [(NSTextField*)id
+		importsGraphics];
+}
+
+void NSTextField_inst_setImportsGraphics_(void *id, BOOL value) {
+	[(NSTextField*)id
+		setImportsGraphics: value];
+}
+
+void* NSTextField_inst_placeholderString(void *id) {
+	return [(NSTextField*)id
+		placeholderString];
+}
+
+void NSTextField_inst_setPlaceholderString_(void *id, void* value) {
+	[(NSTextField*)id
+		setPlaceholderString: value];
+}
+
+void* NSTextField_inst_placeholderAttributedString(void *id) {
+	return [(NSTextField*)id
+		placeholderAttributedString];
+}
+
+void NSTextField_inst_setPlaceholderAttributedString_(void *id, void* value) {
+	[(NSTextField*)id
+		setPlaceholderAttributedString: value];
+}
+
+BOOL NSTextField_inst_allowsDefaultTighteningForTruncation(void *id) {
+	return [(NSTextField*)id
+		allowsDefaultTighteningForTruncation];
+}
+
+void NSTextField_inst_setAllowsDefaultTighteningForTruncation_(void *id, BOOL value) {
+	[(NSTextField*)id
+		setAllowsDefaultTighteningForTruncation: value];
+}
+
+long NSTextField_inst_maximumNumberOfLines(void *id) {
+	return [(NSTextField*)id
+		maximumNumberOfLines];
+}
+
+void NSTextField_inst_setMaximumNumberOfLines_(void *id, long value) {
+	[(NSTextField*)id
+		setMaximumNumberOfLines: value];
+}
+
+double NSTextField_inst_preferredMaxLayoutWidth(void *id) {
+	return [(NSTextField*)id
+		preferredMaxLayoutWidth];
+}
+
+void NSTextField_inst_setPreferredMaxLayoutWidth_(void *id, double value) {
+	[(NSTextField*)id
+		setPreferredMaxLayoutWidth: value];
+}
+
+void* NSTextField_inst_textColor(void *id) {
+	return [(NSTextField*)id
+		textColor];
+}
+
+void NSTextField_inst_setTextColor_(void *id, void* value) {
+	[(NSTextField*)id
+		setTextColor: value];
+}
+
+void* NSTextField_inst_backgroundColor(void *id) {
+	return [(NSTextField*)id
+		backgroundColor];
+}
+
+void NSTextField_inst_setBackgroundColor_(void *id, void* value) {
+	[(NSTextField*)id
+		setBackgroundColor: value];
+}
+
+BOOL NSTextField_inst_drawsBackground(void *id) {
+	return [(NSTextField*)id
+		drawsBackground];
+}
+
+void NSTextField_inst_setDrawsBackground_(void *id, BOOL value) {
+	[(NSTextField*)id
+		setDrawsBackground: value];
+}
+
+BOOL NSTextField_inst_isBezeled(void *id) {
+	return [(NSTextField*)id
+		isBezeled];
+}
+
+void NSTextField_inst_setBezeled_(void *id, BOOL value) {
+	[(NSTextField*)id
+		setBezeled: value];
+}
+
+BOOL NSTextField_inst_isBordered(void *id) {
+	return [(NSTextField*)id
+		isBordered];
+}
+
+void NSTextField_inst_setBordered_(void *id, BOOL value) {
+	[(NSTextField*)id
+		setBordered: value];
+}
+
+BOOL NSTextField_inst_acceptsFirstResponder(void *id) {
+	return [(NSTextField*)id
+		acceptsFirstResponder];
+}
+
+BOOL NSTextField_inst_allowsCharacterPickerTouchBarItem(void *id) {
+	return [(NSTextField*)id
+		allowsCharacterPickerTouchBarItem];
+}
+
+void NSTextField_inst_setAllowsCharacterPickerTouchBarItem_(void *id, BOOL value) {
+	[(NSTextField*)id
+		setAllowsCharacterPickerTouchBarItem: value];
+}
+
+BOOL NSTextField_inst_isAutomaticTextCompletionEnabled(void *id) {
+	return [(NSTextField*)id
+		isAutomaticTextCompletionEnabled];
+}
+
+void NSTextField_inst_setAutomaticTextCompletionEnabled_(void *id, BOOL value) {
+	[(NSTextField*)id
+		setAutomaticTextCompletionEnabled: value];
+}
+
+void* NSTextField_inst_delegate(void *id) {
+	return [(NSTextField*)id
+		delegate];
+}
+
+void NSTextField_inst_setDelegate_(void *id, void* value) {
+	[(NSTextField*)id
 		setDelegate: value];
 }
 
@@ -8246,6 +8461,62 @@ func NSText_alloc() (
 ) {
 	ret := C.NSText_type_alloc()
 	r0 = NSText_fromPointer(ret)
+	return
+}
+
+func NSTextField_alloc() (
+	r0 NSTextField,
+) {
+	ret := C.NSTextField_type_alloc()
+	r0 = NSTextField_fromPointer(ret)
+	return
+}
+
+func NSTextField_labelWithAttributedString_(
+	attributedStringValue core.NSAttributedStringRef,
+) (
+	r0 NSTextField,
+) {
+	ret := C.NSTextField_type_labelWithAttributedString_(
+		objc.RefPointer(attributedStringValue),
+	)
+	r0 = NSTextField_fromPointer(ret)
+	return
+}
+
+func NSTextField_labelWithString_(
+	stringValue core.NSStringRef,
+) (
+	r0 NSTextField,
+) {
+	ret := C.NSTextField_type_labelWithString_(
+		objc.RefPointer(stringValue),
+	)
+	r0 = NSTextField_fromPointer(ret)
+	return
+}
+
+func NSTextField_textFieldWithString_(
+	stringValue core.NSStringRef,
+) (
+	r0 NSTextField,
+) {
+	ret := C.NSTextField_type_textFieldWithString_(
+		objc.RefPointer(stringValue),
+	)
+	r0 = NSTextField_fromPointer(ret)
+	return
+}
+
+func NSTextField_wrappingLabelWithString_(
+	stringValue core.NSStringRef,
+) (
+	r0 NSTextField,
+) {
+	ret := C.NSTextField_type_wrappingLabelWithString_(
+		objc.RefPointer(stringValue),
+	)
+	r0 = NSTextField_fromPointer(ret)
 	return
 }
 
@@ -15111,6 +15382,421 @@ func (x gen_NSText) SetDelegate_(
 	value objc.Ref,
 ) {
 	C.NSText_inst_setDelegate_(
+		unsafe.Pointer(x.Pointer()),
+		objc.RefPointer(value),
+	)
+	return
+}
+
+type NSTextFieldRef interface {
+	Pointer() uintptr
+	Init_asNSTextField() NSTextField
+}
+
+type gen_NSTextField struct {
+	NSControl
+}
+
+func NSTextField_fromPointer(ptr unsafe.Pointer) NSTextField {
+	return NSTextField{gen_NSTextField{
+		NSControl_fromPointer(ptr),
+	}}
+}
+
+func NSTextField_fromRef(ref objc.Ref) NSTextField {
+	return NSTextField_fromPointer(unsafe.Pointer(ref.Pointer()))
+}
+
+func (x gen_NSTextField) SelectText_(
+	sender objc.Ref,
+) {
+	C.NSTextField_inst_selectText_(
+		unsafe.Pointer(x.Pointer()),
+		objc.RefPointer(sender),
+	)
+	return
+}
+
+func (x gen_NSTextField) TextShouldBeginEditing_(
+	textObject NSTextRef,
+) (
+	r0 bool,
+) {
+	ret := C.NSTextField_inst_textShouldBeginEditing_(
+		unsafe.Pointer(x.Pointer()),
+		objc.RefPointer(textObject),
+	)
+	r0 = convertObjCBoolToGo(ret)
+	return
+}
+
+func (x gen_NSTextField) TextShouldEndEditing_(
+	textObject NSTextRef,
+) (
+	r0 bool,
+) {
+	ret := C.NSTextField_inst_textShouldEndEditing_(
+		unsafe.Pointer(x.Pointer()),
+		objc.RefPointer(textObject),
+	)
+	r0 = convertObjCBoolToGo(ret)
+	return
+}
+
+func (x gen_NSTextField) Init_asNSTextField() (
+	r0 NSTextField,
+) {
+	ret := C.NSTextField_inst_init(
+		unsafe.Pointer(x.Pointer()),
+	)
+	r0 = NSTextField_fromPointer(ret)
+	return
+}
+
+func (x gen_NSTextField) IsSelectable() (
+	r0 bool,
+) {
+	ret := C.NSTextField_inst_isSelectable(
+		unsafe.Pointer(x.Pointer()),
+	)
+	r0 = convertObjCBoolToGo(ret)
+	return
+}
+
+func (x gen_NSTextField) SetSelectable_(
+	value bool,
+) {
+	C.NSTextField_inst_setSelectable_(
+		unsafe.Pointer(x.Pointer()),
+		convertToObjCBool(value),
+	)
+	return
+}
+
+func (x gen_NSTextField) IsEditable() (
+	r0 bool,
+) {
+	ret := C.NSTextField_inst_isEditable(
+		unsafe.Pointer(x.Pointer()),
+	)
+	r0 = convertObjCBoolToGo(ret)
+	return
+}
+
+func (x gen_NSTextField) SetEditable_(
+	value bool,
+) {
+	C.NSTextField_inst_setEditable_(
+		unsafe.Pointer(x.Pointer()),
+		convertToObjCBool(value),
+	)
+	return
+}
+
+func (x gen_NSTextField) AllowsEditingTextAttributes() (
+	r0 bool,
+) {
+	ret := C.NSTextField_inst_allowsEditingTextAttributes(
+		unsafe.Pointer(x.Pointer()),
+	)
+	r0 = convertObjCBoolToGo(ret)
+	return
+}
+
+func (x gen_NSTextField) SetAllowsEditingTextAttributes_(
+	value bool,
+) {
+	C.NSTextField_inst_setAllowsEditingTextAttributes_(
+		unsafe.Pointer(x.Pointer()),
+		convertToObjCBool(value),
+	)
+	return
+}
+
+func (x gen_NSTextField) ImportsGraphics() (
+	r0 bool,
+) {
+	ret := C.NSTextField_inst_importsGraphics(
+		unsafe.Pointer(x.Pointer()),
+	)
+	r0 = convertObjCBoolToGo(ret)
+	return
+}
+
+func (x gen_NSTextField) SetImportsGraphics_(
+	value bool,
+) {
+	C.NSTextField_inst_setImportsGraphics_(
+		unsafe.Pointer(x.Pointer()),
+		convertToObjCBool(value),
+	)
+	return
+}
+
+func (x gen_NSTextField) PlaceholderString() (
+	r0 core.NSString,
+) {
+	ret := C.NSTextField_inst_placeholderString(
+		unsafe.Pointer(x.Pointer()),
+	)
+	r0 = core.NSString_fromPointer(ret)
+	return
+}
+
+func (x gen_NSTextField) SetPlaceholderString_(
+	value core.NSStringRef,
+) {
+	C.NSTextField_inst_setPlaceholderString_(
+		unsafe.Pointer(x.Pointer()),
+		objc.RefPointer(value),
+	)
+	return
+}
+
+func (x gen_NSTextField) PlaceholderAttributedString() (
+	r0 core.NSAttributedString,
+) {
+	ret := C.NSTextField_inst_placeholderAttributedString(
+		unsafe.Pointer(x.Pointer()),
+	)
+	r0 = core.NSAttributedString_fromPointer(ret)
+	return
+}
+
+func (x gen_NSTextField) SetPlaceholderAttributedString_(
+	value core.NSAttributedStringRef,
+) {
+	C.NSTextField_inst_setPlaceholderAttributedString_(
+		unsafe.Pointer(x.Pointer()),
+		objc.RefPointer(value),
+	)
+	return
+}
+
+func (x gen_NSTextField) AllowsDefaultTighteningForTruncation() (
+	r0 bool,
+) {
+	ret := C.NSTextField_inst_allowsDefaultTighteningForTruncation(
+		unsafe.Pointer(x.Pointer()),
+	)
+	r0 = convertObjCBoolToGo(ret)
+	return
+}
+
+func (x gen_NSTextField) SetAllowsDefaultTighteningForTruncation_(
+	value bool,
+) {
+	C.NSTextField_inst_setAllowsDefaultTighteningForTruncation_(
+		unsafe.Pointer(x.Pointer()),
+		convertToObjCBool(value),
+	)
+	return
+}
+
+func (x gen_NSTextField) MaximumNumberOfLines() (
+	r0 core.NSInteger,
+) {
+	ret := C.NSTextField_inst_maximumNumberOfLines(
+		unsafe.Pointer(x.Pointer()),
+	)
+	r0 = core.NSInteger(ret)
+	return
+}
+
+func (x gen_NSTextField) SetMaximumNumberOfLines_(
+	value core.NSInteger,
+) {
+	C.NSTextField_inst_setMaximumNumberOfLines_(
+		unsafe.Pointer(x.Pointer()),
+		C.long(value),
+	)
+	return
+}
+
+func (x gen_NSTextField) PreferredMaxLayoutWidth() (
+	r0 core.CGFloat,
+) {
+	ret := C.NSTextField_inst_preferredMaxLayoutWidth(
+		unsafe.Pointer(x.Pointer()),
+	)
+	r0 = core.CGFloat(ret)
+	return
+}
+
+func (x gen_NSTextField) SetPreferredMaxLayoutWidth_(
+	value core.CGFloat,
+) {
+	C.NSTextField_inst_setPreferredMaxLayoutWidth_(
+		unsafe.Pointer(x.Pointer()),
+		C.double(value),
+	)
+	return
+}
+
+func (x gen_NSTextField) TextColor() (
+	r0 NSColor,
+) {
+	ret := C.NSTextField_inst_textColor(
+		unsafe.Pointer(x.Pointer()),
+	)
+	r0 = NSColor_fromPointer(ret)
+	return
+}
+
+func (x gen_NSTextField) SetTextColor_(
+	value NSColorRef,
+) {
+	C.NSTextField_inst_setTextColor_(
+		unsafe.Pointer(x.Pointer()),
+		objc.RefPointer(value),
+	)
+	return
+}
+
+func (x gen_NSTextField) BackgroundColor() (
+	r0 NSColor,
+) {
+	ret := C.NSTextField_inst_backgroundColor(
+		unsafe.Pointer(x.Pointer()),
+	)
+	r0 = NSColor_fromPointer(ret)
+	return
+}
+
+func (x gen_NSTextField) SetBackgroundColor_(
+	value NSColorRef,
+) {
+	C.NSTextField_inst_setBackgroundColor_(
+		unsafe.Pointer(x.Pointer()),
+		objc.RefPointer(value),
+	)
+	return
+}
+
+func (x gen_NSTextField) DrawsBackground() (
+	r0 bool,
+) {
+	ret := C.NSTextField_inst_drawsBackground(
+		unsafe.Pointer(x.Pointer()),
+	)
+	r0 = convertObjCBoolToGo(ret)
+	return
+}
+
+func (x gen_NSTextField) SetDrawsBackground_(
+	value bool,
+) {
+	C.NSTextField_inst_setDrawsBackground_(
+		unsafe.Pointer(x.Pointer()),
+		convertToObjCBool(value),
+	)
+	return
+}
+
+func (x gen_NSTextField) IsBezeled() (
+	r0 bool,
+) {
+	ret := C.NSTextField_inst_isBezeled(
+		unsafe.Pointer(x.Pointer()),
+	)
+	r0 = convertObjCBoolToGo(ret)
+	return
+}
+
+func (x gen_NSTextField) SetBezeled_(
+	value bool,
+) {
+	C.NSTextField_inst_setBezeled_(
+		unsafe.Pointer(x.Pointer()),
+		convertToObjCBool(value),
+	)
+	return
+}
+
+func (x gen_NSTextField) IsBordered() (
+	r0 bool,
+) {
+	ret := C.NSTextField_inst_isBordered(
+		unsafe.Pointer(x.Pointer()),
+	)
+	r0 = convertObjCBoolToGo(ret)
+	return
+}
+
+func (x gen_NSTextField) SetBordered_(
+	value bool,
+) {
+	C.NSTextField_inst_setBordered_(
+		unsafe.Pointer(x.Pointer()),
+		convertToObjCBool(value),
+	)
+	return
+}
+
+func (x gen_NSTextField) AcceptsFirstResponder() (
+	r0 bool,
+) {
+	ret := C.NSTextField_inst_acceptsFirstResponder(
+		unsafe.Pointer(x.Pointer()),
+	)
+	r0 = convertObjCBoolToGo(ret)
+	return
+}
+
+func (x gen_NSTextField) AllowsCharacterPickerTouchBarItem() (
+	r0 bool,
+) {
+	ret := C.NSTextField_inst_allowsCharacterPickerTouchBarItem(
+		unsafe.Pointer(x.Pointer()),
+	)
+	r0 = convertObjCBoolToGo(ret)
+	return
+}
+
+func (x gen_NSTextField) SetAllowsCharacterPickerTouchBarItem_(
+	value bool,
+) {
+	C.NSTextField_inst_setAllowsCharacterPickerTouchBarItem_(
+		unsafe.Pointer(x.Pointer()),
+		convertToObjCBool(value),
+	)
+	return
+}
+
+func (x gen_NSTextField) IsAutomaticTextCompletionEnabled() (
+	r0 bool,
+) {
+	ret := C.NSTextField_inst_isAutomaticTextCompletionEnabled(
+		unsafe.Pointer(x.Pointer()),
+	)
+	r0 = convertObjCBoolToGo(ret)
+	return
+}
+
+func (x gen_NSTextField) SetAutomaticTextCompletionEnabled_(
+	value bool,
+) {
+	C.NSTextField_inst_setAutomaticTextCompletionEnabled_(
+		unsafe.Pointer(x.Pointer()),
+		convertToObjCBool(value),
+	)
+	return
+}
+
+func (x gen_NSTextField) Delegate() (
+	r0 objc.Object,
+) {
+	ret := C.NSTextField_inst_delegate(
+		unsafe.Pointer(x.Pointer()),
+	)
+	r0 = objc.Object_fromPointer(ret)
+	return
+}
+
+func (x gen_NSTextField) SetDelegate_(
+	value objc.Ref,
+) {
+	C.NSTextField_inst_setDelegate_(
 		unsafe.Pointer(x.Pointer()),
 		objc.RefPointer(value),
 	)

--- a/gen/cmd/gen.go
+++ b/gen/cmd/gen.go
@@ -56,6 +56,7 @@ func main() {
 			loadFile("api/appkit/nsstatusbarbutton.objc.json"),
 			loadFile("api/appkit/nsstatusitem.objc.json"),
 			loadFile("api/appkit/nstext.objc.json"),
+			loadFile("api/appkit/nstextfield.objc.json"),
 			loadFile("api/appkit/nstextcontainer.objc.json"),
 			loadFile("api/appkit/nsviewcontroller.objc.json"),
 			loadFile("api/appkit/nsvisualeffectview.objc.json"),

